### DISCUSSION TITLE
How I can set   conditional values of color and icon methods of TextColumn ?

### DISCUSSION THREAD

**Discussion by sergeynilov** [filamentphp/filament#9749] [Help] [answered]
Reading docs 
https://filamentphp.com/docs/3.x/tables/columns/text#customizing-the-color
with examples :
```
TextColumn::make('status')
    ->color('primary')
   ->icon('heroicon-m-envelope')

```   
I did not find if I can set   conditional values of color and icon methods, depeding on status value?
I know how formatStateUsing works, but seems it returns text for TextColumn...

**Comment by devhoussam1998** [comment]
> Reading docs https://filamentphp.com/docs/3.x/tables/columns/text#customizing-the-color with examples :
> 
> ```
> TextColumn::make('status')
>     ->color('primary')
>    ->icon('heroicon-m-envelope')
> ```
> 
> I did not find if I can set conditional values of color and icon methods, depeding on status value? I know how formatStateUsing works, but seems it returns text for TextColumn...

did you check https://filamentphp.com/docs/3.x/support/enums
also check https://www.youtube.com/watch?v=ck722RHs3cc

**Reply by sergeynilov** [reply]
Thank you ! Very interesting, but these examples have only default colors : 'gray', 'warning', 'success', 'danger'.
Can I add custom colors for my app?

**Reply by leandrocfe** [reply, marked as answer]
```php
$panel
    ->colors([
        'my-custom-color' => '#336BFF'
    ])
```
```php
TextColumn::make('status')
->colors([
        'primary' => 'A',
        'my-custom-color' => 'B'
    ])
```
